### ISSUE TITLE
feature: Capability to use commits, PRs, or tags instead of branches to build from

### ISSUE THREAD

**Issue by jes-so** [jdoiro3/mkdocs-multirepo-plugin#154]
The current structure of the building repositories takes only a branch argument. Other determinant arguments are valuable for various workflow situations, e.g. using the "last good SHA" in a branch build failure scenario.

I would like to contribute a non-breaking update where branch behavior is default, but optional arguments can override the default branch.

**Comment by jdoiro3** [comment]
The [import statement](https://github.com/jdoiro3/mkdocs-multirepo-plugin#import-statement) already supports pointing to a specific `tag` since branch is the same as the `git clone` `--branch` option (see [here](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt-code--branchcodeemltnamegtem)). Allowing any commit or a PR number would require some code changes.

**Comment by jes-so** [comment]
You're right, I wrote my issue title clumsily! I meant that it would be good to have options between the four; I'll tweak the issue.

What kinds of changes in this vein would you welcome? I'd obviously like to avoid heavy restructures wherever possible and don't want to change the repo against your vision, but having a fallback SHA would be really valuable for my project. I was thinking about adding an optional commit arg, for example, which would be leveraged to checkout the SHA in sparse_clone.sh. (I included PR as well for this issue, primarily for the sake of completeness, but could break that out.) Would this be feasible?
